angular/projects/common/src/app/search-filter-template/search-filter-template.component.ts

```diff
@@ -0,0 +1,337 @@
+ *
+ * You can add a keyword subset filter by adding a subset object, which will display a dropdown on
+ * the left of the keyword search, allowing for subset searches. See the SubsetObject class for
+ * further instructions on their setup and use
+ *
+ * You can add advanced filters by setting the advancedFilters option to true, and supplying
+ * an array of FilterObjects. Filter objects include definitions for the advanced search
+ * components, their source ID's, selectable options, and type. More informatin is
+ * available in the FilterObject class.
+ *
+ * @export
+ * @class SearchFilterTemplateComponent
+ * @implements {OnInit}
+ * @implements {OnDestroy}
+ */
+@Component({
+  selector: 'search-filter-template',
+  templateUrl: './search-filter-template.component.html',
+  styleUrls: ['./search-filter-template.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.Default
+})
+export class SearchFilterTemplateComponent implements OnInit, OnDestroy {
+  // Inputs
+  @Input() title: string;
+  @Input() tooltip: string;
+  @Input() subsets: SubsetsObject;
+  @Input() advancedFilters = false;
+  @Input() showAdvancedFilters = false;
+  @Input() searchOnFilterChange = true;
+  @Input() filters: FilterObject[] = [];
+
+  // Outputs and Emitters
+  // searchEvent fires whenever a search is executed. The host component is responsible
+  // for parsing the search package and handling as necessary
+  @Output() searchEvent: EventEmitter<any> = new EventEmitter<any>();
+  // resetControls fires when the filter form is reset
+  @Output() resetControls: EventEmitter<void> = new EventEmitter<void>();
+  // filterChange fires whenever a filter on the advanced filter form is changed
+  @Output() filterChange: EventEmitter<any> = new EventEmitter<any>();
+
+  // public vars
+  public FilterType = FilterType; // enum access for the template html
+  public keywordSearchWords: string;
+  public previousKeywords: string;
+  public queryParams = {};
+  public formGroup: FormGroup; // Helper formGroup for grabbing values from controls
+
+  /**
+   * Creates an instance of SearchFilterTemplateComponent.
+   * @param {Router} router Router used to detect URL params
+   * @param {Utils} utils Utils class for date conversion
+   * @memberof SearchFilterTemplateComponent
+   */
+  constructor(
+    private router: Router,
+    public utils: Utils) { }
+
+  ngOnDestroy(): void { }
+
+  ngOnInit() {
+    const urlValues: object = {}; // Storage for the URL params
+
+    // ensure we parse through values from the URL and preselect anything
+    // that needs pre-selecting
+
+    if (this.router && this.router.url) {
+      this.router.url.split(';').forEach(filterVal => {
+        if (filterVal.split('=').length === 2) {
+          const filterName = filterVal.split('=')[0];
+          const val = filterVal.split('=')[1];
+
+          if (val && val !== 'null' && val.length !== 0) {
```

Comment:
`if(val)` accomplishes the same thing here.

Reply:
fixed